docs/docs/security-in-gatsby.md

```diff
@@ -1,9 +1,145 @@
+  // highlight-next-line
+  <input type="hidden" name="comment" value="Hey visit http://maliciouswebsite.com, it's pretty nice" />
+  <input type="submit" />
+</form>
+```
+
+**How can it be prevented?**
+
+### CSRF Tokens
+
+In a page you want to protect, your server will provide an encrypted, hard to guess **token**, which is tied to the user's session and thus will be required to be sent back in the POST form:
+
+```js
+<form action="http://mywebsite.com/blog/addcoment" method="POST">
+  <input type="text" name="comment" />
+  // highlight-next-line
+  <input type="hidden" name="token" value="SU9J3tMoT1w5q6uJ1VMXaaf9UXzLvyNd" />
+  <input type="submit" />
+</form>
+```
+
+So when the form is sent, the server will compare them and block the action if they are not the same. Note that usually the malicious website won't have access to this CSRF token because of [HTTP Access Control](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Access-Control-Allow-Origin).
+
+### Same-Site Cookies Directive
+
+If you need to create cookies in your application, make sure to protect them by adding the `SameSite` directive:
+
+`Set-Cookie: example=1; SameSite=Strict`
+
+It allows the server to make sure that the cookies are not being sent by a **cross-site** domain request.
+Check out [MDN Docs](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Set-Cookie) regarding how a cookie could be configured. [Most moderns browsers support it with a backward compatibility](https://caniuse.com/#feat=same-site-cookie-attribute).
+
+## Third-party Scripts
+
+Some third-party scripts like Google Tag Manager gives the possibility to add arbitrary JavaScript code on top of them as a [custom HTML tag](https://support.google.com/tagmanager/answer/6107167), which can be a malicious script. To avoid that, be sure to have a [access control](https://support.google.com/tagmanager/answer/6107011) management in these services.
+
+## Check Your Dependencies
+
+In your Gatsby project, you are going to have a lot of dependencies (which has their own dependencies as well) in your `node_modules/`. Therefore, it is important to check if any of them has a security issue.
+
+**Using `npm`**
+
+In npm, you can use the `npm audit` command to check your dependencies. It is available since v6. Check their [docs](https://docs.npmjs.com/cli/audit) for more options.
+
+**Using `yarn`**
+
+Similar to `npm`, you can use the `yarn audit` command. It is available since v1.12.0, not available in v2. Check their [docs](https://classic.yarnpkg.com/en/docs/cli/audit/) for more options.
+
+## Key Security
+
+Gatsby provides the capability of [fetching data from anywhere](/docs/content-and-data/) and by that you should know that most of this data requires a key to access them. These keys should be stored in the server by using [Environment Variables](/docs/environment-variables/) as this example below, fetching data from GitHub with an Authentication Header:
```

Comment:
```suggestion
Gatsby provides the capability of [fetching data from various APIs](/docs/content-and-data/) and those APIs often require a key to access them. These keys should be stored in the build environment using [Environment Variables](/docs/environment-variables/) as shown below when fetching data from GitHub with an Authentication Header:
```

Reply:
Also a link to the appropriate section of https://www.gatsbyjs.org/docs/environment-variables/ would be good.